Sources/MongoSwift/BSON/Document.swift

```diff
@@ -395,8 +417,11 @@ extension Document {
                 if let newValue = newValue {
                     try self.setValue(for: key, to: newValue)
                 } else {
-                    // TODO: SWIFT-224: use va_list variant of bson_copy_to_excluding to improve performance
-                    self = self.filter { $0.key != key }
+                    guard let bson = bson_new() else {
+                        fatalError("Invalid BSON data")
+                    }
+                    withVaList([]) { bson_copy_to_excluding_noinit_va(self._bson, bson, key, $0) }
```

Comment:
with my proposed change below, we could probably route this through the helper as well.

Reply:
Done.